src/dftbp/api/mm/mmapi.F90

```diff
@@ -634,6 +648,101 @@ subroutine TDftbPlus_getCM5Charges(this, atomCharges)
   end subroutine TDftbPlus_getCM5Charges
 
 
+  !> Get the reference atomic charges for the atoms of the system to be neutral
+  subroutine TDftbPlus_getRefCharges(this, z0)
+
+    !> Instance
+    class(TDftbPlus), intent(in) :: this
+
+    !> Atomic valence reference charges
+    real(dp), intent(out) :: z0(:)
+
+    real(dp), allocatable :: q0(:, :, :)
+    integer :: mOrb, nAtom, nSpin
+
+    call this%checkInit()
+
+    if (this%main%uniqHubbU%mHubbU > 1) then
+      call error("Reference charge call unsupported for shell resolved models")
+    end if
+    mOrb = this%main%orb%mOrb
+    nAtom = nrOfAtoms(this%main)
+    nSpin = this%main%nSpin
+    allocate(q0(mOrb, nAtom, nspin), source=0.0_dp)
```

Comment:
Is it really necessary to initialize with zeros here?

Reply:
Not really, but again the pattern is becoming common. The call below is intent out, so it's definitely irrelevant (I'll remove)
```suggestion
    allocate(q0(mOrb, nAtom, nspin))
```